crates/config/src/providers/remappings.rs

```diff
@@ -30,11 +30,18 @@ impl Remappings {
         Self { remappings }
     }
 
+    fn filter_key(r: &Remapping) -> String {
+        match &r.context {
+            Some(str) => str.clone() + &r.name.clone(),
+            None => r.name.clone(),
+        }
+    }
+
     /// Consumes the wrapper and returns the inner remappings vector.
     pub fn into_inner(self) -> Vec<Remapping> {
         let mut tmp = HashSet::new();
         let remappings =
-            self.remappings.iter().filter(|r| tmp.insert(r.name.clone())).cloned().collect();
+            self.remappings.iter().filter(|r| tmp.insert(Self::filter_key(&r))).cloned().collect();
```

Comment:
I wonder if this is actually sufficient filtering for remappings? cc @mattsse 

I think we'd love to see the result of using this in bigger, more complex repos, albeit, on its own, this essentially allows users to insert remapping contexts _manually_, which is nice because it makes them entirely opt in. A bit hesitant about this and we'd need to run some good QA, but, somewhat supportive.